Jenkinsfile

```diff
@@ -31,38 +33,39 @@ pipeline {
     stage('Checkout') {
       options { skipDefaultCheckout() }
       steps {
+        deleteDir()
         gitCheckout(basedir: "${BASE_DIR}")
-        stash allowEmpty: true, name: 'source', useDefaultExcludes: false
-        script {
-          env.GO_VERSION = readFile("${BASE_DIR}/.go-version").trim()
-          env.BUILD_FILEBEAT = isChanged(["^filebeat/.*"])
-          env.BUILD_HEARTBEAT = isChanged(["^heartbeat/.*"])
-          env.BUILD_AUDITBEAT = isChanged(["^auditbeat/.*"])
-          env.BUILD_METRICBEAT = isChanged(["^metricbeat/.*"])
-          env.BUILD_PACKETBEAT = isChanged(["^packetbeat/.*"])
-          env.BUILD_WINLOGBEAT = isChanged(["^winlogbeat/.*"])
-          env.BUILD_DOCKERLOGBEAT = isChanged(["^x-pack/dockerlogbeat/.*"])
-          env.BUILD_FUNCTIONBEAT = isChanged(["^x-pack/functionbeat/.*"])
-          env.BUILD_JOURNALBEAT = isChanged(["^journalbeat/.*"])
-          env.BUILD_GENERATOR = isChanged(["^generator/.*"])
-          env.BUILD_KUBERNETES = isChanged(["^deploy/kubernetes/*"])
-          env.BUILD_DOCS = isGitRegionMatch(patterns: ["^docs/.*"], comparator: 'regexp') || params.runAllStages
-          env.BUILD_LIBBEAT = isGitRegionMatch(patterns: ["^Libbeat/.*"], comparator: 'regexp') || params.runAllStages
+        dir("${BASE_DIR}"){
+          script {
+            env.GO_VERSION = readFile(".go-version").trim()
+            env.BUILD_FILEBEAT = isChanged(["^filebeat/.*"])
+            env.BUILD_HEARTBEAT = isChanged(["^heartbeat/.*"])
+            env.BUILD_AUDITBEAT = isChanged(["^auditbeat/.*"])
+            env.BUILD_METRICBEAT = isChanged(["^metricbeat/.*"])
+            env.BUILD_PACKETBEAT = isChanged(["^packetbeat/.*"])
+            env.BUILD_WINLOGBEAT = isChanged(["^winlogbeat/.*"])
+            env.BUILD_DOCKERLOGBEAT = isChanged(["^x-pack/dockerlogbeat/.*"])
+            env.BUILD_FUNCTIONBEAT = isChanged(["^x-pack/functionbeat/.*"])
+            env.BUILD_JOURNALBEAT = isChanged(["^journalbeat/.*"])
+            env.BUILD_GENERATOR = isChanged(["^generator/.*"])
+            env.BUILD_KUBERNETES = isChanged(["^deploy/kubernetes/*"])
+            env.BUILD_DOCS = isGitRegionMatch(patterns: ["^docs/.*"], comparator: 'regexp') || params.runAllStages
+            env.BUILD_LIBBEAT = isGitRegionMatch(patterns: ["^libbeat/.*","^x-pack/filebeat./*"], comparator: 'regexp') || params.runAllStages
```

Comment:
I think we actually need a separate condition for all the x-pack builds where we have both OSS and X-Pack.

```
env.BUILD_WINLOGBEAT = isChanged(["^winlogbeat/.*"])
env.BUILD_XPACK_WINLOGBEAT = isChanged(["^winlogbeat/.*", "^x-pack/winlogbeat/.*"])

env.BUILD_LIBBEAT = isChanged(["^libbeat/.*"])
env.BUILD_XPACK_LIBBEAT = isChanged(["^libbeat/.*", "^x-pack/libbeat/.*"])
```

The X-Pack code depends on the OSS so you should probably build/test the X-Pack side when OSS changes. But the opposite is not true since the dependency only goes in one direction.

Reply:
If there is a change in `libbeat` everything that use `isChanged` function is build 

```
def isChanged(patterns){
  return (params.runAllStages
    || isGitRegionMatch(patterns: patterns, comparator: 'regexp')
    || isGitRegionMatch(patterns: ["^libbeat/.*"], comparator: 'regexp')
  )
}
```

I am not sure which modules would trigger `BUILD_XPACK_WINLOGBEAT` or `BUILD_XPACK_LIBBEAT `

Reply:
> If there is a change in libbeat everything that use isChanged function is build

That's good (I didn't check what that function what doing). This is roughly what the dependency tree looks like.

<img width="466" alt="Screen Shot 2020-01-31 at 10 24 57 AM" src="https://user-images.githubusercontent.com/4565752/73551352-4243dd80-4414-11ea-8625-935efd73e166.png">

Could you try to codify the triggers to be a little closer to that. Let me know if you want a little more clarity on the dependencies.

> I am not sure which modules would trigger BUILD_XPACK_WINLOGBEAT or BUILD_XPACK_LIBBEAT

So for example BUILD_XPACK_WINLOGBEAT would trigger on changes to libbeat, x-pack/libbeat, winlogbeat, or x-pack/winlogbeat. And BUILD_XPACK_LIBBEAT would trigger on changes to libbeat or x-pack/libbeat.